### PR TITLE
add kaitai_struct_cpp_stl_runtime/0.9

### PR DESCRIPTION
**Pull request by aapocketz**
Specify library name and version:  **kaitai_struct_cpp_stl_runtime/0.9**

I am not the author of the library, and the library has no special dependencies. This is the runtime C++ parser dependency for kaitai structs https://kaitai.io/. It does not include other tools for other languages or the code generator stages for kaitai (which generates parsers for domain specific language type markup). Its a reasonably well used tool by other open source projects like Kismet (wireless packet sniffer tool https://www.kismetwireless.net/) and I believe it would be useful as a library for Conan center. I have included only the most recent version 0.9.

This is my first conan-center contribution so any feedback is greatly appreciated. 

(edit fixed wrong email typo updating pull request)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution.

Take a look on my review.

Zlib is optional, so please, add it as requirement. https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/blob/0.9/CMakeLists.txt#L6

iconv is options, add it as optional too: https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/blob/0.9/CMakeLists.txt#L8

**Comment by aapocketz** [comment]
Ok, I see that conan doesn't like shared=True being a default option. The upstream library seems to only want to generate a shared library, I don't see an obvious way to make a static library. I am not a cmake expert so correct me if wrong.
https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/blob/master/CMakeLists.txt

How are packages like this handled?

**Comment by uilianries** [comment]
@aapocketz Remove both options `shared` and `fPIC`. Also, remove `config_options` and `configure` methods. It doesn't make keeping fPIC, as it's mandatory for shared library. If there is no option for static library, doesn't make sense keeping it optional.

File: recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py
Comment (uilianries) [inline recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py:24]:
```suggestion
    short-paths = True
```

Because Windows build failed due maximum path length.

https://docs.conan.io/en/latest/reference/conanfile/attributes.html#short-paths

**Comment by N-Nagorny** [comment]
@aapocketz, I guess, I know the reason why tests building fails on Windows. I created an issue and a pull request to the repo of the library (shown right above this comment). Until it's merged, you may want to patch the existing CMakeLists.txt with `tools.replace_in_file()`. If you don't have time, let me know. I think, I can make this change on your branch or on a new one copied from yours.

**Comment by N-Nagorny** [comment]
@uilianries, is this possible to merge a recipe with git commit instead of versioned tag? I think, the build issues are solved in https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/pull/40 so this branch can be appropriate to continue work on the PR.

**Comment by N-Nagorny** [comment]
@aapocketz, @uilianries, build fix is [delivered](https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/commit/cb09b3a84c7e905c6a3fecfe9617cf864d9af781) to `master` of `kaitai_struct_cpp_stl_runtime` repo. Is this possible to point `conandata.yml` to this exact commit of `master` and retry build?

**Comment by aapocketz** [comment]
I don't believe there is a way, at least not well documented, to point to a commit in master, it has to be a released tarball. I think we could include a set of patches ontop of the last release though as an option. Or we can wait until a new release. Not sure I am not really involved in either conan or kaitai directly.

File: recipes/kaitai_struct_cpp_stl_runtime/all/conandata.yml
Comment (N-Nagorny) [inline recipes/kaitai_struct_cpp_stl_runtime/all/conandata.yml:4]:
```suggestion
sources:
  "cb09b3a84c7e905c6a3fecfe9617cf864d9af781":
    url: "https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/archive/cb09b3a84c7e905c6a3fecfe9617cf864d9af781.zip"
    sha256: "058c01828692bf193cb862372651c572fcc43babbf2d664674409511b9f54e5b"
```

File: recipes/kaitai_struct_cpp_stl_runtime/config.yml
Comment (N-Nagorny) [inline recipes/kaitai_struct_cpp_stl_runtime/config.yml:2]:
```suggestion
  "cb09b3a84c7e905c6a3fecfe9617cf864d9af781":
```

**Comment by N-Nagorny** [comment]
@aapocketz, take a look at the suggestions please. It works locally and it would be good to commit them at least to trigger CI for recipe and code check.

**Comment by N-Nagorny** [comment]
@aapocketz, thanks.
@uilianries, any other changes required before merging?

File: recipes/kaitai_struct_cpp_stl_runtime/all/conandata.yml
Comment (uilianries) [inline recipes/kaitai_struct_cpp_stl_runtime/all/conandata.yml:2]:
```suggestion
  "cb09b3a84c7e905c6a3fecfe9617cf864d9af781":
```

Use the official version 0.9. Or, ask to the author for a new tag, 0.8 is from 2020. Or, use the date, not the commit: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases

File: recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py
Comment (uilianries) [inline recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py:43]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version],
                  destination=self._source_subfolder, strip_root=True)
```

File: recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py
Comment (uilianries) [inline recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py:57]:
```suggestion
        self._cmake.definitions["STRING_ENCODING_TYPE"] = "ICONV" if self.options.with_iconv else "NONE"
```

File: recipes/kaitai_struct_cpp_stl_runtime/all/conandata.yml
Comment (N-Nagorny) [inline recipes/kaitai_struct_cpp_stl_runtime/all/conandata.yml:2]:
```suggestion
sources:
  "cci.20210701":
```

File: recipes/kaitai_struct_cpp_stl_runtime/config.yml
Comment (N-Nagorny) [inline recipes/kaitai_struct_cpp_stl_runtime/config.yml:2]:
```suggestion
versions:
  "cci.20210701":
```

**Review by N-Nagorny — APPROVED** [review]
@aapocketz, could you take a look at the suggested changes please?

**Comment by aapocketz** [comment]
@N-Nagorny I don't think that worked, I think the right way to do it would be to either add a patch to the 0.9 release, or get like a 1.0 tag or some other release tagged version?

**Comment by N-Nagorny** [comment]
@aapocketz, it works (I tried it locally) with the changes applied and changes **uilianries** suggested above.

**Comment by N-Nagorny** [comment]
@uilianries, could you take a look please? Does it rather look like a CI or package issue?

File: recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py
Comment (prince-chrismc) [inline recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py:42]: duplicate line

```suggestion
```

File: recipes/kaitai_struct_cpp_stl_runtime/all/test_package/conanfile.py
Comment (prince-chrismc) [inline recipes/kaitai_struct_cpp_stl_runtime/all/test_package/conanfile.py:7]: unused

```suggestion
    generators = "cmake"
```

File: recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py
Comment (prince-chrismc) [reply recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py:58]: Should there be an option for zlib?

**Review by prince-chrismc — COMMENTED** [review]
You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

File: recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py
Comment (uilianries) [reply recipes/kaitai_struct_cpp_stl_runtime/all/conanfile.py:58]: Nope, there is no option for zlib. It's only find_package https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/blob/master/CMakeLists.txt#L35

**Comment by N-Nagorny** [comment]
@aapocketz, looks like it works now! Could you accept the last 3 suggested changes from [here](https://github.com/conan-io/conan-center-index/pull/5132/files) to keep `conanfile.py` clean?

**Comment by aapocketz** [comment]
done!

**Comment by N-Nagorny** [comment]
Hi, @uilianries! Looks like all review suggestions are covered. Are any other changes required before merging?

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**
